### PR TITLE
Migrate all plugins to girder 3.x

### PR DESCRIPTION
**Pull request by psavery**
This is still a work in progress. The girder-server can load the
molecules plugin without errors, but the interactive web api docs
do not load.

Signed-off-by: Patrick Avery <patrick.avery@kitware.com>

### CODE REVIEW

**Comment by psavery** [comment]
Nevermind, the web api docs DO load. I just had to build the client with the new `girder build` command.

I can test it more, and I need to update the pytests with the new api as well.

**Comment by psavery** [comment]
Tests now pass for the molecules plugin

**Comment by psavery** [comment]
@cjh1 Now that girder uses pip and setup.py for plugins and plugin dependencies, there are potentials for name conflicts. For instance, if I put in that the queues plugin depends on `cumulus`, it will probably automatically download the `cumulus` package from PyPI, which is an Amazon Web Services package (and not the girder plugin we are looking for), if cumulus is not already installed. 

`molecules` is a package on PyPI already (it is a machine learning for molecular dynamics package).

I'm wondering if we should have some specific naming conventions for these. For now, I called the molecules package `girder-molecules`. The package it provides is `molecules`, but I should maybe change that to `girder_molecules`. I called some of the packages things like `openchemistryapp`, etc.

**Comment by cryos** [comment]
Let's favor openchemistry as a namespace for our plugin code @psavery

**Comment by psavery** [comment]
@cryos Will do.

**Comment by psavery** [comment]
So, I think this may already be ready for others to review. I don't know all of the use cases we want to test, but everything works as it did before in my basic testing via the interactive web API docs. I can create molecules, delete them, GET them, etc.

It's a little easier to install as well, since you don't even need to clone girder to install the plugin. With all of the system dependencies already, I can install it like this:

```
virtualenv -p python3 mongochemserver
source ./mongochemserver/bin/activate

mkdir ./mongochemserver/src
cd ./mongochemserver/src
git clone -b migrate-to-girder-3.x https://github.com/openchemistry/mongochemserver

cd mongochemserver/girder/molecules
pip install -e . # Will install girder as well

girder build # Build the interactive web api
girder-server # log in and enable the molecules plugin
```

I also did a few small things with openchemistrypy, and my small tests worked.

**Comment by cryos** [comment]
The best test would be whether this can replace the previous girder and serve a JupyterLab notebook in the same way the previous version could. So set the full stack up as it was, run a few calculations, visualize the electronic structure. Then replace the Girder server with your ported branch and see if you can do the same. Bonus points for the single page application (should work fine if the JupyterLab widgets work).

**Comment by psavery** [comment]
@cryos @cjh1 After my most recent changes, I tried out the notebooks Alessandro pushed with both girder-2.x and girder-3.x. And I was able to run them successfully for both! I ran the lines on the nwchem notebook, the chemml notebook, and the psi4 notebook.

The only thing that wasn't working for me was visualizing the vibrational modes in the nwchem notebook. That didn't work for either girder-2.x and girder-3.x for me (I would get a pop-up error after some time - maybe 5 minutes - that would something like `Request Entity Too Large`). But visualizing the vibrational modes with psi4 worked fine for me for both.

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,13 @@
-FROM girder/girder:2.x-maintenance-py3
+FROM girder/girder:latest-py3
 
 # Install cumulus
-RUN git clone https://github.com/Kitware/cumulus.git /cumulus && \
+RUN git clone -b migrate-to-girder-3.x https://github.com/Kitware/cumulus.git /cumulus && \
```
Comment (psavery) [inline docker/Dockerfile]: We will remove `-b migrate-to-girder-3.x` once the cumulus branch gets merged.

**Pull request merged**